pkg/util/fedinformer/genericmanager/single-cluster-manager.go

```diff
@@ -86,12 +87,18 @@ type singleClusterInformerManagerImpl struct {
 }
 
 func (s *singleClusterInformerManagerImpl) ForResource(resource schema.GroupVersionResource, handler cache.ResourceEventHandler) {
+	// Add a RLock to make sure the transform func is set before starting
+	s.lock.RLock()
 	// if handler already exist, just return, nothing changed.
-	if s.IsHandlerExist(resource, handler) {
+	if s.isHandlerExist(resource, handler) {
+		s.lock.RUnlock()
 		return
 	}
 
-	s.informerFactory.ForResource(resource).Informer().AddEventHandler(handler)
+	i := s.informerFactory.ForResource(resource).Informer()
+	_ = i.SetTransform(fedinformer.StripUnusedFields)
```

Comment:
Maybe you can look at https://github.com/karmada-io/karmada/pull/2383.

Reply:
I think adding a `RLock` is enough to make sure the transform func is set before starting, please see https://github.com/karmada-io/karmada/pull/2089#discussion_r948568485

Reply:
I mean the `i.SetTransform(fedinformer.StripUnusedFields)` might be called multiple times for a specific `resource type`, we don't need to set the transform function every time.
See https://github.com/karmada-io/karmada/blob/71d037be0bd006d8c43f24b50ac65855c4ad5d8c/pkg/util/fedinformer/typedmanager/single-cluster-manager.go#L183-L188.

Reply:
Do you mean adding an extra map to check if the informer has been transformed/started? If so I think they take the same effect but it would use extra memory to maintain the map

Reply:
You can leave this file behind(untouched), because this is probably more involved than doing the transforming thing. Maybe we can ask @Poor12 to make a subsequent PR for this.

Reply:
> You can leave this file behind(untouched)

OK, have reverted this part